framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/AMethodWrapper.java

```diff
@@ -0,0 +1,149 @@
+     * "java.lang.Object" if the return type is unknown. The type may contain generic type
+     * arguments.
+     */
+    private String returnType = "java.lang.Object";
+
+    /**
+     * A mirror of the parameters field of AMethod, but using AFieldWrapper objects as the values.
+     * Keys are parameter indices. Like the parameters field of AMethod, this map starts empty and
+     * is vivified by calls to {@link #vivifyParameter(int, String, Name)} or {@link
+     * #vivifyParameter(int, TypeMirror, Name)}.
+     */
+    private Map<Integer, AFieldWrapper> parameters = new HashMap<>();
+
+    /** The type parameters of this method. */
+    private List<? extends TypeParameterElement> typeParameters;
+
+    /**
+     * Wrap an AMethod. Package-private, because it should only be called from AClassWrapper.
+     *
+     * @param theMethod the method to wrap
+     * @param methodElt the method's declaration
+     */
+    AMethodWrapper(AMethod theMethod, ExecutableElement methodElt) {
+        this.theMethod = theMethod;
+        this.setReturnType(methodElt.getReturnType().toString());
+        this.vivifyParameters(methodElt);
+        this.typeParameters = methodElt.getTypeParameters();
+    }
+
+    /**
+     * The return type, in a format that can be printed in Java source code.
+     *
+     * @return the return type as a string using fully-qualified names (in the style of {@link
+     *     TypeMirror#toString()}), or "java.lang.Object" if the return type is unknown
+     */
+    public String getReturnType() {
+        return returnType;
+    }
+
+    /**
+     * Get the type parameters of this method.
+     *
+     * @return the list of type parameters
+     */
+    public List<? extends TypeParameterElement> getTypeParameters() {
+        return typeParameters;
+    }
+
+    /**
+     * Provide the AMethodWrapper with a return type.
+     *
+     * @param returnType a string representation of the type, in the form returned by {@link
+     *     TypeMirror#toString()}
+     */
+    private void setReturnType(String returnType) {
+        if ("java.lang.Object".equals(this.returnType)) {
+            this.returnType = returnType;
+        }
+    }
+
+    /**
+     * Populates the method parameter map for the method. This is called from the constructor, so
+     * that the method parameter map always has an entry for each parameter.
+     *
+     * @param methodElt the method whose parameters should be vivified
+     */
+    private void vivifyParameters(ExecutableElement methodElt) {
+        for (int i = 0; i < methodElt.getParameters().size(); i++) {
+            VariableElement ve = methodElt.getParameters().get(i);
+            TypeMirror type = ve.asType();
+            Name name = ve.getSimpleName();
+            this.vivifyParameter(i, type, name);
+        }
+    }
+
+    /**
+     * Add the given parameter to the scene-lib representation.
```

Comment:
This describes a possible side effect that doesn't necessarily occur.  It doesn't describe the return value.
The documentation of the various `vivify*` methods is inconsistent.  Please make them consistent.  I like the way the `ASceneWrapper` ones are worded.